core/controllers/profile.py

```diff
@@ -587,15 +587,28 @@ class UrlHandler(base.BaseHandler):
 
     GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
 
+    URL_PATH_ARGS_SCHEMAS = {}
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {
+            'current_url': {
+                'schema': {
+                        'type': 'basestring'
+                 },
+                 'default_value': None
```

Comment:
@vojtechjelinek because in some cases, current_url is not mentioned as parameter while using UrlHandler. Then, it is throwing an error. So, I have added default value to be None.

Reply:
Where it is not mentioned? (I'm mainly interested in the frontend usages)

Reply:
![default_issue](https://user-images.githubusercontent.com/78922463/153243306-08ff543e-ad3b-4095-b2e9-ac8f1b7f45de.png)
I am getting this error while running the test without setting default as None. @vojtechjelinek Can you please tell me whether this screenshot is enough to prove that "current_url" is not mentioned somewhere ??

Reply:
No, you need to check the frontend usage of this handler.

Reply:
Hey @vojtechjelinek , I am checking for the part where it is mentioned in frontend. But I couldn't understand where my CI check failed because I have ran "**python -m scripts.run_backend_tests --test_shard 1**" command and it showed me that all the tests passed successfully without any error. Can you please help me with this issue why I have got this CI check failed ? 
Thank you!

Reply:
Not sure what you mean? The CI checks seem to be passing.

Reply:
![CIfailed](https://user-images.githubusercontent.com/78922463/153418698-5616b6b5-4a39-4d93-9771-23223fee792e.png)
I am new to use these terms like Continuous Integration and Continuous Deployment (CI/CD). So, I am not sure what this exactly meant but I am getting that some checks are not succesful. I have read [this](https://github.com/oppia/oppia/wiki/If-CI-checks-fail-on-your-PR#failing-tests-and-lint-checks) too but I couldn't find the answer for my problem.
What can I do to make it a successful check ?

Reply:
Ah, e2e tests are sometimes flaky, I will restart them.

Reply:
> No, you need to check the frontend usage of this handler.

https://user-images.githubusercontent.com/78922463/153483939-1a5e99cc-f541-4912-8a4a-f0c7ef0928e8.mp4

Hello @vojtechjelinek This was the case where we can get that error. There is current_url query parameter mentioned everywhere other than this case which is manually going to /url_handler. So, what should I do now ??

Reply:
> You will need to check the frontend files that are in the codebase, and provide that we don't set `current_url`.

![searchUrl](https://user-images.githubusercontent.com/78922463/153642841-9106da62-9b48-4c43-9516-be499d14fa25.png)
 
Hello @vojtechjelinek , I have searched every file where url_handler is being mentioned and the only cases where it is being called without using `current_url` is only in test files. So, we can't see them in frontend files (I have confirmed by searching the method name used in test files and that method is called nowhere) Now, should I change the test files so that we can not get error while running the tests or I have to do something else ?

Reply:
Yep change the tests.

Reply:
The thing is that we want to make sure that the schema validation is as strict as possible, it will also help simplify the code below. It is pretty common to modify tests, they just make sure that the code follows some rules that we want to follow. Here the rules in that we want to make the `current_url` not `None` and because of that we need to modify the tests.